core/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java

```diff
@@ -63,7 +64,7 @@ public boolean processExistingRecoveries(RoutingAllocation allocation) {
         boolean changed = false;
         MetaData metaData = allocation.metaData();
         RoutingNodes routingNodes = allocation.routingNodes();
-        List<Tuple<ShardRouting, UnassignedInfo>> recoveriesToCancel = new ArrayList<>();
+        List<Runnable> recoveriesToCancel = new ArrayList<>();
```

Comment:
The name for this is a little weird, since it's not recoveries to cancel, it's more like "cancellations to run". When I see `List<Runnable> recoveriesToCancel` I expect some sort of code like:

```java
for (Runnable task : recoveriesToCancel) {
    task.cancel();
}
```

Instead, the `.run()` method does the actual cancellation, so maybe the var name could be changed to be more clear?